examples/v1beta1/pipelineruns/pipelinerun-with-when-expressions.yaml

```diff
@@ -1,109 +1,97 @@
-          operator: in
-          values: ["yes"]
-      taskSpec:
-        steps:
-          - name: echo
-            image: ubuntu
-            script: 'echo file exists'
-    - name: echo-file-missing
-      when:
-        - input: "$(tasks.check-file.results.exists)"
-          operator: in
-          values: ["missing"]
-      taskSpec:
-        steps:
-          - name: echo
-            image: ubuntu
-            script: 'echo file missing'
-    - name: task-should-be-skipped
-      when:
-        - input: "$(params.path)"
-          operator: notin
-          values: ["README.md"]
-      taskSpec:
-        steps:
-          - name: echo
-            image: ubuntu
-            script: exit 1
----
-apiVersion: tekton.dev/v1beta1
-kind: PipelineRun
-metadata:
-  name: guarded-pr
-spec:
-  serviceAccountName: 'default'
-  pipelineRef:
-    name: guarded-pipeline
+            workspace: source
+        runAfter:
+          - create-file
+        taskSpec:
+          params:
+            - name: path
+          workspaces:
+            - name: source
+              description: The workspace to check for the file
+          results:
+            - name: exists
+              description: indicates whether the file exists or is missing
+          steps:
+            - name: check-file
+              image: alpine
+              script: |
+                if test -f $(workspaces.source.path)/$(params.path); then
+                  printf yes | tee /tekton/results/exists
+                else
+                  printf no | tee /tekton/results/exists
+                fi
+      - name: echo-file-exists # when expression using task result, evaluates to true
+        when:
+          - input: "$(tasks.check-file.results.exists)"
+            operator: in
+            values: ["yes"]
+        taskSpec:
+          steps:
+            - name: echo
+              image: ubuntu
+              script: 'echo file exists'
+      - name: task-should-be-skipped-1
+        when:
+          - input: "$(tasks.check-file.results.exists)" # when expression using task result, evaluates to false
+            operator: in
+            values: ["missing"]
+        taskSpec:
+          steps:
+            - name: echo
+              image: ubuntu
+              script: exit 1
+      - name: task-should-be-skipped-2 # when expression using parameter, evaluates to false
+        when:
+          - input: "$(params.path)"
+            operator: notin
+            values: ["README.md"]
+        taskSpec:
+          steps:
+            - name: echo
+              image: ubuntu
+              script: exit 1
   params:
     - name: path
       value: README.md
   workspaces:
     - name: source
-      volumeClaimTemplate:
-        spec:
-          accessModes:
-          - ReadWriteOnce
-          resources:
-            requests:
-              storage: 1Gi
+      persistentVolumeClaim:
+        claimName: shared-task-storage
```

Comment:
i think we should keep the original volumeClaimTemplate here since it makes the example easier to run